packages/interactivity-router/src/index.ts

```diff
@@ -370,10 +370,17 @@ if ( globalThis.IS_GUTENBERG_PLUGIN ) {
 		document.addEventListener(
 			'click',
 			function ( event ) {
-				const ref = ( event.target as Element ).closest( 'a' );
-				if ( isValidLink( ref ) && isValidEvent( event ) ) {
+				const target = event.target as Element;
+				const ref = target.closest( 'a' );
+				if (
+					isValidLink( ref ) &&
+					isValidEvent( event ) &&
+					! target.hasAttribute( 'data-wp-on--click' ) // Don't override other click directives.
+				) {
 					event.preventDefault();
 					actions.navigate( ref.href );
```

Comment:
We should await here, otherwise the `scrollToTop` will happen before the navigation is complete.

Reply:
Oh, that makes sense 🙂 

On the other hand, as `actions.navigate` is a generator, I am not sure what is the best approach to wait for it.

I assume we can't just use await, right? Do we have to iterate through all the yields and check once they have the `done` property?

Reply:
It's not a generator, it's an async function 😄

The async function is created on the fly by the store proxy: 

https://github.com/WordPress/gutenberg/blob/0d1943cf84e60ad4646333ed68eca7cc7b879009/packages/interactivity/src/store.ts#L108-L109

Reply:
Okay 😄 I'm still a bit lost on that part and Visual Studio is complaining because "'await' has no effect on the type of this expression.".

I just pushed a commit with the await: [link](https://github.com/WordPress/gutenberg/pull/62342/commits/ae9a75088e26524a88a3242b992ed659742a2cd7).

Reply:
> 'await' has no effect on the type of this expression

Oh, maybe we need to improve the types of the actions 🙂

Reply:
I have opened a pull request to improve the types and convert actions into async functions.

- https://github.com/WordPress/gutenberg/pull/62400